### PR TITLE
Update graphql-tools to v7 scoped packages and simplify dependency

### LINKED ISSUE #4923: AS3: Consider upgrading to the latest graphql-tools
Apollo Server uses `graphql-tools` to convert from typeDefs/resolvers/schemaDirectives/parseOptions to a schema, and to implement mocks/mockEntireSchema.  (We also re-export all of its symbols from all the integration packages, but we've already removed that from AS3.) We currently depend on `graphql-tools@4. They moved on to v5 last April (and also split the package into more packages). We should probably update to the latest stable version and depend on the sub-packages (`@graphql-tools/schema` and `@graphql-tools/mock`).

Alternatively we should consider whether graphql-tools is necessary or whether there's a more direct way to implement AS's API (or if the API should change). Probably we just want to upgrade though!

### PR DESCRIPTION
**Pull request by glasser**
Note that a previous commit on `release-3.0` stopped exporting all
`graphql-tools` exports from our packages, but the docs had not yet been
updated.

- Update from `graphql-tools@4` to the newest version of various
  `@graphql-tools/X` packages.
- Make it clear that the use of `graphql-tools` functionality in Apollo
  Server is essentially a convenience for the most common use cases, and
  that folks who want to use detailed `graphql-tools` functionality are
  welcome to do so directly.
- Remove support for `schemaDirectives`. This `makeExecutableSchema`
  option is considered to be legacy by `graphql-tools` these days, so
  this seems like a reasonable time to drop it from the Apollo Server
  API. Folks who want to keep using it can use `makeExecutableSchema`
  from `@graphql-tools/schema` directly, or ideally switch to the newer
  `schemaTransforms`. Stop documenting it.
- Remove most docs that just duplicate `graphql-tools` docs. (Keep some
  of the mocking docs.)
- Update mocking support to use `addMocksToSchema` from
  `@graphql-tools/mock` rather than `addMockFunctionsToSchema` from
  `graphql-tools`. Note that this has some backwards-incompatible
  changes, mentioned in the CHANGELOG. Update docs to not claim you can
  look at arguments in mock functions (that's one of those changes).
  Document alternative to deprecated `MockList` class. Remove skipped
  test because presumably the "bug" is by this point expected behavior.
- Only apply mocks to schema in the non-`gateway` codepath rather than
  to schemas provided by the gateway (at startup or later via
  `onSchemaChange`). Assuming your `gateway` is an `ApolloGateway`, this
  should be a no-op as `ApolloGateway` provides an `executor` which
  never calls resolvers on the schema. If you have implemented your own
  `gateway` which does not have an executor that entirely ignores
  resolvers, you can always add mocks to the schema returned from `load`
  and sent to `onSchemaChange` yourself.

Inspired by #4237 by @yaacovCR. Fixes #4923.

### CODE REVIEW

**Comment by glasser** [comment]
In the interest of velocity (eg, I want to make some alphas today) I'm going to merge this without review but I *definitely* want feedback on this soon!

File: docs/source/testing/mocking.md [hunk 1]
```diff
@@ -86,65 +86,23 @@ const mocks = {
...
   Person: () => ({
-    // a list of length between 2 and 6 (inclusive)
-    friends: () => new MockList([2,6]),
-    // a list of three lists each with two items: [[1, 1], [2, 2], [3, 3]]
-    listOfLists: () => new MockList(3, () => new MockList(2)),
+    // a list of length between 2 and 6, using the "casual" npm module
+    // to generate a random integer
+    friends: [...new Array(casual.integer(2, 6))],
+    // a list of three lists of two items: [[1, 1], [2, 2], [3, 3]]
+    listOfLists: () => [...new Array(3)].map((i) => [...new Array(2)]),
```
Comment (trevor-scheer) [inline docs/source/testing/mocking.md:102]: Noticed this on a drive-by
```suggestion
    listOfLists: () => [...new Array(3)].map((_, i) => [i, i]),
```

Comment (glasser) [reply docs/source/testing/mocking.md:102]: I adapted this from the [graphql-tools docs](https://www.graphql-tools.com/docs/mocking/#using-lists-in-mocks) (which I suspect were adapted from these docs in the first place) FWIW

**Review by abernix — COMMENTED** [review]
LookedGTM.
